src/item_factory.cpp

```diff
@@ -1899,10 +1899,12 @@ void Item_factory::load( islot_magazine &slot, const JsonObject &jo, const std::
 {
     bool strict = src == "dda";
     if( jo.has_array( "ammo_type" ) ) {
-        for( const std::string id : jo.get_array( "ammo_type" ) ) {
+        slot.type.clear();
```

Comment:
You are calling `slot.type.clear();` in both branches, so you can just move it outside of `if-else `block.